storage/postgres/storage_test.go

```diff
@@ -11,35 +11,165 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
 package postgres
 
 import (
+	"bytes"
 	"context"
+	"crypto"
+	"crypto/sha256"
 	"database/sql"
 	"flag"
+	"fmt"
 	"os"
 	"testing"
-	"time"
 
 	"github.com/golang/glog"
+	"github.com/google/trillian"
+	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/storage"
 	"github.com/google/trillian/storage/postgres/testdb"
+	storageto "github.com/google/trillian/storage/testonly"
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/types"
 )
 
-// db is shared throughout all postgres tests
-var db *sql.DB
+func TestNodeRoundTrip(t *testing.T) {
+	cleanTestDB(db, t)
+	tree := createTreeOrPanic(db, storageto.LogTree)
+	s := NewLogStorage(db, nil)
+
+	const writeRevision = int64(100)
+	nodesToStore := createSomeNodes()
+	nodeIDsToRead := make([]storage.NodeID, len(nodesToStore))
+	for i := range nodesToStore {
+		nodeIDsToRead[i] = nodesToStore[i].NodeID
+	}
+
+	{
+		runLogTX(s, tree, t, func(ctx context.Context, tx storage.LogTreeTX) error {
+			forceWriteRevision(writeRevision, tx)
+
+			// Need to read nodes before attempting to write
+			if _, err := tx.GetMerkleNodes(ctx, 99, nodeIDsToRead); err != nil {
+				t.Fatalf("Failed to read nodes: %s", err)
+			}
+			if err := tx.SetMerkleNodes(ctx, nodesToStore); err != nil {
+				t.Fatalf("Failed to store nodes: %s", err)
+			}
+			return nil
+		})
+	}
+
+	{
+		runLogTX(s, tree, t, func(ctx context.Context, tx storage.LogTreeTX) error {
+			readNodes, err := tx.GetMerkleNodes(ctx, 100, nodeIDsToRead)
+			if err != nil {
+				t.Fatalf("Failed to retrieve nodes: %s", err)
+			}
+			if err := nodesAreEqual(readNodes, nodesToStore); err != nil {
+				t.Fatalf("Read back different nodes from the ones stored: %s", err)
+			}
+			return nil
+		})
+	}
+}
+func forceWriteRevision(rev int64, tx storage.TreeTX) {
+	mtx, ok := tx.(*logTreeTX)
+	if !ok {
+		panic(nil)
+	}
+	mtx.treeTX.writeRevision = rev
+}
+
+func createSomeNodes() []storage.Node {
+	r := make([]storage.Node, 4)
+	for i := range r {
+		r[i].NodeID = storage.NewNodeIDWithPrefix(uint64(i), 8, 8, 8)
+		h := sha256.Sum256([]byte{byte(i)})
+		r[i].Hash = h[:]
+		glog.Infof("Node to store: %v\n", r[i].NodeID)
```

Comment:
No trailing newline required for log messages.
```suggestion
		glog.Infof("Node to store: %v", r[i].NodeID)
```